File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.js
function solution(k, tangerine) {
    const MAP = new Map();
    for(const size of tangerine) {
        MAP.set(size, (MAP.get(size) || 0) + 1);
    }
    const sizes = [...MAP].map(([v, s]) => s).sort((a, b) => b - a);
    let count = 0;
    let answer = 0;
    for(const size of sizes) {
        if(count + size <= k) {
            count += size;
            answer++;
        } else {
            if(count < k) answer++;
            break;
        }
    }
    
    return answer;
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
from collections import Counter

def solution(k, tangerine):
    counter = Counter(tangerine)
    answer = total = 0
    sorted_invent = sorted(counter.values(), reverse=True)
    for cnt in sorted_invent:
        if total >= k:
            return answer
        answer += 1
        total += cnt
    return answer